src/ray/gcs/gcs_server/test/redis_gcs_table_storage_test.cc

```diff
@@ -0,0 +1,506 @@
+                                                        const ClientID &node_id) {
+    rpc::TaskReconstructionData data;
+    data.set_task_id(task_id.Binary());
+    data.set_num_reconstructions(1);
+    data.set_node_manager_id(node_id.Binary());
+    return data;
+  }
+
+  rpc::ObjectTableDataList GenObjectTableDataList(const ClientID &node_id) {
+    rpc::ObjectTableDataList data;
+    rpc::ObjectTableData object_table_data;
+    object_table_data.set_manager(node_id.Binary());
+    object_table_data.set_object_size(1);
+    data.add_items()->CopyFrom(object_table_data);
+    return data;
+  }
+
+  rpc::GcsNodeInfo GenGcsNodeInfo(const ClientID &node_id) {
+    rpc::GcsNodeInfo data;
+    data.set_node_id(node_id.Binary());
+    data.set_state(rpc::GcsNodeInfo_GcsNodeState_ALIVE);
+    return data;
+  }
+
+  rpc::ResourceMap GenResourceMap(const ClientID &node_id) {
+    rpc::ResourceTableData resource_table_data;
+    resource_table_data.set_resource_capacity(1.0);
+    rpc::ResourceMap data;
+    (*data.mutable_items())["attr1"] = resource_table_data;
+    return data;
+  }
+
+  rpc::HeartbeatTableData GenHeartbeatTableData(const ClientID &node_id) {
+    rpc::HeartbeatTableData data;
+    data.set_client_id(node_id.Binary());
+    return data;
+  }
+
+  rpc::HeartbeatBatchTableData GenHeartbeatBatchTableData(const ClientID &node_id) {
+    rpc::HeartbeatBatchTableData data;
+    rpc::HeartbeatTableData heartbeat_table_data;
+    heartbeat_table_data.set_client_id(node_id.Binary());
+    data.add_batch()->CopyFrom(heartbeat_table_data);
+    return data;
+  }
+
+  rpc::ErrorTableData GenErrorInfoTable(const JobID &job_id) {
+    rpc::ErrorTableData data;
+    data.set_job_id(job_id.Binary());
+    return data;
+  }
+
+  rpc::ProfileTableData GenProfileTableData() {
+    rpc::ProfileTableData data;
+    data.set_component_type("object_manager");
+    return data;
+  }
+
+  rpc::WorkerFailureData GenWorkerFailureData() {
+    rpc::WorkerFailureData data;
+    data.set_timestamp(std::time(nullptr));
+    return data;
+  }
+
+  template <typename TABLE, typename KEY, typename VALUE>
+  void Put(TABLE &table, const KEY &key, const VALUE &value) {
+    auto on_done = [this](Status status) { --pending_count_; };
+    ++pending_count_;
+    RAY_CHECK_OK(table.Put(key, value, on_done));
+    WaitPendingDone();
+  }
+
+  template <typename TABLE, typename KEY, typename VALUE>
+  int Get(TABLE &table, const KEY &key, std::vector<VALUE> &values) {
+    auto on_done = [this, &values](Status status, const boost::optional<VALUE> &result) {
+      RAY_CHECK_OK(status);
+      --pending_count_;
+      values.clear();
+      if (result) {
+        values.push_back(*result);
+      }
+    };
+    ++pending_count_;
+    RAY_CHECK_OK(table.Get(key, on_done));
+    WaitPendingDone();
+    return values.size();
+  }
+
+  template <typename TABLE, typename KEY, typename VALUE>
+  int GetAll(TABLE &table, std::vector<std::pair<KEY, VALUE>> &values) {
```

Comment:
It seems that `GetAll`, `GetByJobId` and `DeleteByJobId` are not tested?

Reply:
Redis store client does not support AsyncGetAll, AsyncGetByIndex and AsyncDeleteByIndex now. I have remove GetAll, GetByJobId and DeleteByJobId.